activerecord/lib/active_record/store.rb

```diff
@@ -136,6 +166,36 @@ def store_accessor_for(store_attribute)
         type_for_attribute(store_attribute.to_s).accessor
       end
 
+      class AccessorType # :nodoc
+        def self.type_for_accessor(type_name)
+          case type_name
+          when Class
+            type_name.new
+          when Symbol
+            klass = standard_types.fetch(type_name, Type::Value)
+            klass.new
+          else
+            type_name
+          end
+        end
+
+        private
+
+        def self.standard_types
```

Comment:
Yeah. It's late, and I'll look at this more in the morning, but I'd be against this feature overall and definitely this should lean in that direction implementation wise.